### ISSUE TITLE
Add a Github Actions job testing Intel compiler

### ISSUE THREAD

**Issue by GiovanniBussi** [plumed/plumed2#677]
It looks like intel C++ compiler and intel MPI are now free, so they could be tested.

[Here](https://software.intel.com/content/www/us/en/develop/articles/installing-intel-oneapi-toolkits-via-apt.html) is the documentation on how to install them with apt-get.

**Comment by GiovanniBussi** [comment]
I tested with Docker and the test failed. [Here](https://github.com/GiovanniBussi/plumed2/commit/cbc45b95b61308461ed3bf184296a7f370bacdd7) is the commit. The error was:
```
Step 1/6 : FROM intel/oneapi-basekit
latest: Pulling from intel/oneapi-basekit
f22ccc0b8772: Pulling fs layer
3cf8fb62ba5f: Pulling fs layer
e80c964ece6a: Pulling fs layer
d9a33df66fd4: Pulling fs layer
165b45d43311: Pulling fs layer
701867559230: Pulling fs layer
d9a33df66fd4: Waiting
165b45d43311: Waiting
701867559230: Waiting
3cf8fb62ba5f: Verifying Checksum
3cf8fb62ba5f: Download complete
e80c964ece6a: Download complete
f22ccc0b8772: Verifying Checksum
f22ccc0b8772: Download complete
701867559230: Verifying Checksum
701867559230: Download complete
f22ccc0b8772: Pull complete
3cf8fb62ba5f: Pull complete
e80c964ece6a: Pull complete
d9a33df66fd4: Verifying Checksum
d9a33df66fd4: Download complete
d9a33df66fd4: Pull complete
165b45d43311: Verifying Checksum
165b45d43311: Download complete
failed to register layer: Error processing tar file(exit status 1): write /opt/intel/oneapi/intelpython/python3.7/lib/python3.7/site-packages/xgboost/lib/libxgboost.so: no space left on device
make: *** [Makefile:11: intel] Error 1
make: Leaving directory '/home/runner/work/plumed2/plumed2/docker'
```
I guess the container takes too much space.

**Comment by GiovanniBussi** [comment]
I managed to install intel compilers correctly in GitHub Actions. Unfortunately, several tests fail for numerical reasons. See [this branch](https://github.com/plumed/plumed2/tree/test-intel). I will not merge it

**Comment by GiovanniBussi** [comment]
This is fixed in #694
